### PR TITLE
Refactor, env vars, and update packages

### LINKED ISSUE #587: Update JS Packages
Update yarn packages to latest version without any breaking changes.

### LINKED ISSUE #604: Update hasura from v1.3.3 to v2.0.2
Hasura is outdate, is very import to upgrade to have lastest stable changes. Check it out to update to [v2.0.2](https://hasura.io/blog/migrating-from-hasura-v1-3-to-v2-0/)

### LINKED ISSUE #613: Update docker-compose.staging.yml

### LINKED ISSUE #618: prefix for env vars
standardize prefix for env vars based on service name

`POSTGRES_`
`HASURA_`
`HAPI_`
`WEBAPP_`

### PR DESCRIPTION
**Pull request by xavier506**
### GH Issue

resolve #587 
resolve #613
resolve #618
resolve #604

Chart.js breaking changes specification [here](https://www.chartjs.org/docs/next/getting-started/v3-migration.html#specific-changes)

Eslintrc bug fix [here](https://github.com/codefactor-io/codefactor/issues/21)

### CODE REVIEW

**Review by AngeloCG97 — APPROVED** [review]

**Comment by leisterfrancisco** [comment]
@tetogomez @adriexnet @AngeloCG97 

Localhost:
image

Production:
image

After updating `chart.js` and `react-chartjs-2` It came with breaking changes that after fixing them produced some UI differences any suggestion?

File: Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ ARG branch
 RUN rm /bin/sh && ln -s /bin/bash /bin/sh
 
 # Get the packages needed for Ubuntu
-ENV DEBIAN_FRONTEND noninteractive
+ENV DEBIAN_webapp noninteractive
```
Comment (leisterfrancisco) [inline Dockerfile:11]: Shouln't weapp be uppercase?

**Review by leisterfrancisco — COMMENTED** [review]
@xavier506 thanks very much about file restructuring, looks so nice now. 💯

File: docker-compose.yml [hunk 2]
```diff
@@ -20,42 +20,42 @@ services:
...
-      POSTGRES_PORT: "${POSTGRES_PORT}"
-      POSTGRES_DB: "${POSTGRES_DB}"
-      POSTGRES_HOST: "${POSTGRES_HOST}"
-      POSTGRES_SCHEMA: "${POSTGRES_SCHEMA}"
-      EOS_API_ENDPOINT: "${EOS_API_ENDPOINT}"
+      HAPI_POSTGRES_USER: "${POSTGRES_USER}"
+      HAPI_POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
+      HAPI_POSTGRES_PORT: "${POSTGRES_PORT}"
+      HAPI_POSTGRES_DB: "${POSTGRES_DB}"
+      HAPI_POSTGRES_HOST: "${HAPI_POSTGRES_HOST}"
```
Comment (leisterfrancisco) [inline docker-compose.yml:44]: @xavier506  Wasn't it `POSTGRES_HOST`?

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (xavier506) [reply docker-compose.yml]: @leisterfrancisco  we need this to be  `HAPI_POSTGRES_HOST`

`POSTGRES_HOST = localhost`
 `HAPI_POSTGRES_HOST = eosrate_postgres `

`localhost` only works inside the postgres pod, other services need to resolve a hostname such as `eosrate_postgres`

File: .env.example [hunk 3]
```diff
@@ -25,13 +25,13 @@ HASURA_GRAPHQL_UNAUTHORIZED_ROLE=anonymous
...
-REACT_APP_EOS_API_URL=https://jungle.eosio.cr
-REACT_APP_EOS_API_HOST=jungle.eosio.cr
-REACT_APP_EOS_API_PORT=443
-REACT_APP_EOS_API_PROTOCOL=https
-REACT_APP_EOS_CHAIN_ID=2a02a0053e5a8cf73a56ba0fda11e4d92e0238a4a2aa74fccf46d5a910746840
-REACT_APP_BLOCK_EXPLORER=https://jungle3.bloks.io
-REACT_APP_NETWORK_MONITOR_URL=https://jungle.eosio.online/
\ No newline at end of file
+WEBAPP_GRAPHQL_HTTP_URL=http://localhost:8080/v1/graphql
+WEBAPP_API_URL=https://jungle.eosio.cr
```
Comment (leisterfrancisco) [inline .env.example:29]: @xavier506 should I remove it, right? is not been used except by `webapp/Dockerfile`

File: webapp/readme.md [hunk 4]
```diff
@@ -18,7 +18,7 @@ We use [standardjs code style](https://standardjs.com) and we favor [functional
 |     |── language/ .................................... i18n translations
 |     |── models/ ...................................... rematch models
 |     |── routes/ ...................................... route components
-|     |── services/ .................................... singleton services ( vanillajs )
+|     |──  .................................... singleton services ( vanillajs )
```
[the remark this answers is not in the document]
Comment (leisterfrancisco) [reply webapp/readme.md:21]: removed

**Pull request merged**